README.md
# Text-speach-ai
Text speech AI for free which uses eleven labs API with advanced Python and you can use it without limitation 


<p align="center">
  <img src="https://img.shields.io/badge/Version-3.0-green?style=for-the-badge">
  <img src="https://img.shields.io/github/license/kamanati/gmphish?style=for-the-badge">
  <img src="https://img.shields.io/github/stars/kamanati/gmphish?style=for-the-badge">
  <img src="https://img.shields.io/github/issues/kamanati/gmphish?color=red&style=for-the-badge">
  <img src="https://img.shields.io/github/forks/kamanati/gmphish?color=teal&style=for-the-badge">
</p>

<p align="center">
  <img src="https://img.shields.io/badge/Author-hasan--fq-cyan?style=flat-square">
  <img src="https://img.shields.io/badge/Open%20Source-Yes-cyan?style=flat-square">
  <img src="https://img.shields.io/badge/Written%20In-Bash-cyan?style=flat-square">
</p>

<p align="center">A Tool to create Real life voice  </p>


## features ##
```
[-] you can access elevenlabs vouced for free wiithout limit restrictions
[-] All features like stability anf boost vouices are availabe
[-] for more use the tool 🤌

```
# Example Audio #

[Audio.webm](https://cdn.whyp.it/5dd2c38d-9afe-44f3-8c55-dbccdb049b39.mp3)

## Examples ##
```
python xmain.py --text "Your Text" --output file.np3
```
### From file ###
```
python xmain.py --file FILE.txt --line 1 # use --line 001 for full file in one task
```
### Output Result 
```
Task (14): "Love is the music that makes life's journey sweeter. Dance to its rhythm. Let love fill your heart with joy. Love's melody brings joy to life's dance."

Task completed (14): "Love is the music that makes life's journey sweeter. Dance to its rhythm. Let love fill your heart with joy. Love's melody brings joy to life's dance."

Metadata and image added successfully!
Task (15): "Success is not about reaching the top, but about enjoying the climb. Find joy in the process. Each step of your journey is meaningful. Success is in the journey itself."

Task completed (15): "Success is not about reaching the top, but about enjoying the climb. Find joy in the process. Each step of your journey is meaningful. Success is in the journey itself."

Metadata and image added successfully!
```
it go through the next line as a task 
### note ###
The audio files are stored in directory audio in 'Nami' in random formate like 
Motive_f7fe5cc1.wav

## Usage ##
```
python xmain.py -h
usage: xmain.py [-h] [-t TEXT] [-o OUTPUT] [-f FILE] [-l LINE]
                [-v VOICE] [-s STABEL] [-b BOOST]
                [--voices-list] [--cheat] [--api-key API_KEY]

Generate and save audio from text or file

options:
  -h, --help            show this help message and exit
  -t TEXT, --text TEXT  Text to generate audio from
  -o OUTPUT, --output OUTPUT
                        Output audio file name
  -f FILE, --file FILE  File in txt form (task: one per line)
  -l LINE, --line LINE  Starting line number (default: 1)
  -v VOICE, --voice VOICE
                        Name of the voice (default: 'Adam')
  -s STABEL, --stabel STABEL
                        Voice stability MAX:0.99(default: 0.75)
  -b BOOST, --boost BOOST
                        Voice stability boost MAX:0.99(default:
                        0.80)
  --voices-list         List available voices in database
  --cheat               display cheat to bypass limitaion
                        restriction(only for legal)
  --api-key API_KEY     Set the API key

```



### Find Me on :
<p align="left">
  <a href="https://github.com/Kamanati/zphisher" target="_blank"><img src="https://img.shields.io/badge/Github-kamanati-zph-green?style=for-the-badge&logo=github"></a>
  <a href="https://www.instagram.com/hasanfq6" target="_blank"><img src="https://img.shields.io/badge/IG-%40hasanfq6-red?style=for-the-badge&logo=instagram"></a>
  <a href="https://chat.whatsapp.com/9629013836" target="_blank"><img src="https://img.shields.io/badge/Chat-whatsapp-blue?style=for-the-badge&logo=whatsapp"></a>
</p>


Audio/News
thanks for using my tool


1sample.py
import eyed3
import argparse

def set_audio_metadata(audio_file, title, artist, album, image_file):
    try:
        audio = eyed3.load(audio_file)
        if audio.tag is None:
            audio.initTag()
        
        audio.tag.title = title
        audio.tag.artist = artist
        audio.tag.album = album
        
        # Embed the image as cover art
        if image_file:
            with open(image_file, "rb") as image:
                audio.tag.images.set(eyed3.id3.frames.ImageFrame.FRONT_COVER, image.read(), "image/jpeg")
        
        audio.tag.save()
        print("Metadata and image added successfully!")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Set metadata and add an image to an audio file")
    parser.add_argument("--files", help="Path to the audio file")
    parser.add_argument("--title", help="Title of the audio")
    parser.add_argument("--artist", help="Artist of the audio")
    parser.add_argument("--album", help="Album name of the audio")
    parser.add_argument("--image", help="Path to the image file (cover art)")
    
    args = parser.parse_args()
    
    if args.files:
        audio_file = args.files
        title = args.title or "Nami's speech"
        artist = args.artist or "Hasanfq6"
        album = args.album or "Motivational speech"
        image_file = args.image
        
        set_audio_metadata(audio_file, title, artist, album, image_file)
    else:
        print("Please provide the path to the audio file using the --files argument.")


xmain.py
import argparse,subprocess,uuid,eyed3
from time import sleep
from elevenlabs import *

#TODO Adam: pNInz6obpgDQGcFmaJgB
#TODO Clyde: 2EiwWnXFnvU5JabPnv8n

# Function to read voice mappings from the voices.txt file
def read_voice_mappings(file_path):
    voice_mappings = {}
    try:
        with open(file_path, 'r') as voices_file:
            for line in voices_file:
                parts = line.strip().split(': ')
                if len(parts) == 2:
                    voice_name, voice_id = parts[0], parts[1]
                    voice_mappings[voice_name] = voice_id
    except FileNotFoundError:
        print("Voices file (voices.txt) not found.")
    return voice_mappings

# Define the path to the voices.txt file
voices_file_path = 'voices.txt'

# Read voice mappings from voices.txt
voice_id_mapping = read_voice_mappings(voices_file_path)

def generate_and_save_audio(text, voice, output_file):
    # Generate audio from the provided text with the specified voice
    audio = generate(text=text, voice=voice)

    # Save the audio to the specified output file
    with open(output_file, "wb") as audio_file:
        audio_file.write(audio)

def list_voices():
    print("Available voices:")
    for voice_name in voice_id_mapping:
        print(voice_name)

def main():
    parser = argparse.ArgumentParser(description="Generate and save audio from text or file")
    parser.add_argument("-t", "--text", help="Text to generate audio from")
    parser.add_argument("-o", "--output", help="Output audio file name")
    parser.add_argument("-f", "--file", help="File in txt form (task: one per line)")
    parser.add_argument("-l", "--line", default="1", help="Starting line number (default: 1)")
    parser.add_argument("-v", "--voice", default="Clyde", help="Name of the voice (default: 'Adam')")
    parser.add_argument("-s", "--stabel", type=float, default=0.75, help="Voice stability MAX:0.99(default: 0.75)")
    parser.add_argument("-b", "--boost", type=float, default=0.80, help="Voice stability boost MAX:0.99(default: 0.80)")
    parser.add_argument("--voices-list", action="store_true", help="List available voices in database")
    parser.add_argument("--cheat", action="store_true", help="display cheat to bypass limitaion restriction(only for legal)")
    parser.add_argument("--api-key", help="Set the API key")

    args = parser.parse_args()

    # Set the API key if provided
    if args.api_key:
        set_api_key(args.api_key)
        print(f"API key set: {args.api_key}")
        return
    if args.cheat:
        # Run cheat.py using subprocess
        subprocess.run(["python", "cheat.py"])
        return

    if args.voices_list:
        list_voices()
        return

    if args.file:
        with open(args.file, 'r') as file:
            lines = file.readlines()
            for line_number, line in enumerate(lines, start=1):
                if line_number < int(args.line):
                    continue  # Skip lines before the specified starting line

                input_text = line.strip()
                if not input_text:
                    print("Empty task, skipping.")
                    continue

                if args.voice:
                    if args.voice in voice_id_mapping:
                        voice_id = voice_id_mapping[args.voice]
                    else:
                        print(f"Voice '{args.voice}' not found in voices.txt. Using default voice ID.")
                        voice_id = "2EiwWnXFnvU5JabPnv8n"  # Set the default voice ID
                else:
                    print("No voice specified. Using default voice ID.")
                    voice_id = "2EiwWnXFnvU5JabPnv8n"  # Set the default voice ID

                selected_voice = Voice(
                    name=args.voice,
                    category='premade',
                    settings=VoiceSettings(stability=args.stabel, similarity_boost=args.boost),
                    voice_id=voice_id
                )

                print(f"Task ({line_number}): {input_text}")
                sleep(8)
                words = input_text.split()
                segments = [words[i:i + 37] for i in range(0, len(words), 37)]

                for segment_number, segment in enumerate(segments, start=1):
                    segment_text = " ".join(segment)
                    segment_output_file = f"segment_{line_number}_{segment_number}.wav"
                    generate_and_save_audio(segment_text, selected_voice, segment_output_file)

                combined_audio = b""
                for segment_number in range(1, len(segments) + 1):
                    segment_output_file = f"segment_{line_number}_{segment_number}.wav"
                    with open(segment_output_file, "rb") as segment_audio_file:
                        combined_audio += segment_audio_file.read()
                output_file = f"Audio/Motive_pri_{str(uuid.uuid4())[:8]}{line_number}.wav"
#                output_file = f"output_{line_number}.wav"
                with open(output_file, "wb") as combined_audio_file:
                    combined_audio_file.write(combined_audio)
                print(f"\n\033[32mTask completed (\033[36m{line_number}\033[32m)\033[0m: {input_text}\n")
                os.system(f"python 1sample.py --file {output_file} --image Audio/1.png")
    elif args.text and args.output:
        # ... (rest of your code)
            input_text = args.text
            output_file = args.output
        
            if args.voice:
                if args.voice in voice_id_mapping:
                    voice_id = voice_id_mapping[args.voice]
                else:
                    print(f"Voice '{args.voice}' not found in voices.txt. Using default voice ID.")
                    voice_id = "2EiwWnXFnvU5JabPnv8n"  # Set the default voice ID
            else:
                print("No voice specified. Using default voice ID.")
                voice_id = "2EiwWnXFnvU5JabPnv8n"  # Set the default voice ID
        
            selected_voice = Voice(
                name=args.voice,
                category='premade',
                settings=VoiceSettings(stability=args.stabel, similarity_boost=args.boost),
                voice_id=voice_id
            )
        
            print(f"{input_text}")
            sleep(8)
            words = input_text.split()
            segments = [words[i:i + 37] for i in range(0, len(words), 37)]
        
            for segment_number, segment in enumerate(segments, start=1):
                segment_text = " ".join(segment)
                segment_output_file = f"segment_{segment_number}.wav"
                generate_and_save_audio(segment_text, selected_voice, segment_output_file)
        
            combined_audio = b""
            for segment_number in range(1, len(segments) + 1):
                segment_output_file = f"segment_{segment_number}.wav"
                with open(segment_output_file, "rb") as segment_audio_file:
                    combined_audio += segment_audio_file.read()
        
            with open(output_file, "wb") as combined_audio_file:
                combined_audio_file.write(combined_audio)
            print(f"Combined audio saved as '{output_file}'")
            os.system(f"python 1sample.py --file {output_file} --image Audio/1.png")
    else:
        print("Please provide either --file or both --text and --output, -h.")

if __name__ == main():
  try:
    main()
  except KeyError:
    print("voice error")
